velox/functions/sparksql/SimpleComparisonChecker.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/lib/SimpleComparisonMatcher.h"
+
+namespace facebook::velox::functions::sparksql {
+using namespace facebook::velox::functions;
+
+class SparkComparisonMatcher : public Matcher {
+ public:
+  SparkComparisonMatcher(
+      const std::string& prefix,
+      std::vector<MatcherPtr> inputMatchers,
+      std::string* op)
+      : prefix_{prefix}, inputMatchers_{std::move(inputMatchers)}, op_{op} {
+    VELOX_CHECK_EQ(2, inputMatchers_.size());
+  }
+
+  bool match(const core::TypedExprPtr& expr) override {
+    if (auto call = dynamic_cast<const core::CallTypedExpr*>(expr.get())) {
+      const auto& name = call->name();
+      if (name == prefix_ + "equalto" || name == prefix_ + "lessthan" ||
+          name == prefix_ + "greaterthan") {
+        if (allMatch(call->inputs(), inputMatchers_)) {
+          *op_ = name;
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+ private:
+  const std::string prefix_;
+  std::vector<MatcherPtr> inputMatchers_;
+  std::string* op_;
+};
+
+class SparkSimpleComparisonChecker : public SimpleComparisonChecker {
```

Comment:
It seems the only difference between SparkSimpleComparisonChecker and PrestoSimpleComparisonChecker is the names. I wonder if we can define some pure virtual functions in the base class to return function names and override them for Spark and Presto in the derived classes. In this way, the other parts of code can be reused.